openpype/hosts/maya/api/lib.py

```diff
@@ -934,6 +934,7 @@ def namespaced(namespace, new=True):
 
     """
     original = cmds.namespaceInfo(cur=True, absoluteName=True)
+    original_relative_names = cmds.namespace(query=True, relativeNames=True)
     if new:
         namespace = unique_namespace(namespace)
         cmds.namespace(add=namespace)
```

Comment:
I think you're missing the lines that actually set the `relative_names` state before the context too.

This is what I shared with you on discord:
```python
@contextlib.contextmanager
def namespaced(namespace, new=True, relative_names=None):
    """Work inside namespace during context

    Args:
        new (bool): When enabled this will rename the namespace to a unique
            namespace if the input namespace already exists.
        relative_names (bool): Whether to force relative names to True or
            False. Defaults to None to not change the current state.

    Yields:
        str: The namespace that is used during the context

    """
    original = cmds.namespaceInfo(cur=True, absoluteName=True)
    original_relative_names = cmds.namespace(query=True, relativeNames=True)
    if new:
        namespace = unique_namespace(namespace)
        cmds.namespace(add=namespace)
    if relative_names is not None:
        cmds.namespace(relativeNames=relative_names)

    try:
        cmds.namespace(set=namespace)
        yield namespace
    finally:
        cmds.namespace(set=original)
        if relative_names is not None:
            cmds.namespace(relativeNames=original_relative_names)
```

Note how before the yield we're also applying the relative names.